Doc 472/custom pack registry updates (#1217)
spectrocloud/librarium: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md

Please ensure that the ports 443 and 80 are exclusively allocated to the registry server and are not in use by other processes.
</WarningBox>

> # Deploying a Pack Registry Server with letsencrypt certificates

Author: yes.

> # Deploying a Pack Registry Server with letsencrypt certificates

Reviewer: ```suggestion
# Deploy a Pack Registry Server with SSL Certificates
```
Just trying to think of ways to shorten the title.

Author: Adapted a bit. Hope the new title is better.

The following section demonstrates how you can deploy a registry server secured with a TLS certificate issued by [Let's Encrypt](https://letsencrypt.org/).

> For more advanced configuration and deployment with self-signed certificates see [advanced configuration.](/registries-and-packs/advanced-configuration)

Reviewer: ```suggestion

Check out the [advanced configuration](/registries-and-packs/adding-a-custom-registry#self-signedcertificates) section for more advanced configuration and deployment with self-signed certificates.
```
Added a newline and changed the order to improve the flow.

Author: Fixed.

Palette provides a Docker image for the pack registry server.
The following steps need to be performed to deploy the pack registry server using this docker image:

> 1. Create an httppasswd file 

Reviewer: ```suggestion
1. Create a folder that contains an httppasswd file. 
```

Author: Fixed.

```

> mkdir spectropaxconfig

Reviewer: This step reads better if split it out. Here is a previous example. 

![CleanShot 2023-03-29 at 16 21 49](https://user-images.githubusercontent.com/29551334/228689415-bfcf9f68-d26e-4e86-b126-20a1ac3abe2b.png)

htpasswd -Bbn admin {enter your secure password choice} > spectropaxconfig/htpasswd-basic
```

> 2. Create a pax registry configuration
> 
> Edit and save in `spectropaxconfig/myconfig.yml`

Reviewer: ```suggestion
2. Create a pax registry configuration file titled **myconfig.yml**. This file should be in the **spectropaxconfig** directory.
```

```yaml
version: 0.1
log:
  level: debug
storage: inmemory
http:
  addr: :5000
  tls:
    letsencrypt:
      cachefile: /etc/spectropaxconfig/le-cache
      email: you@companydomain.com
      hosts: 
      - yourhost.companydomain.com 
auth:
  htpasswd:
    realm: basic-realm
    path: /etc/spectropaxconfig/htpasswd-basic
```

> 3. Run the container image with the following flags

Reviewer: ```suggestion
3. Start the container image with the following flags.
```

```
docker run  \
    --rm \
    -p 443:5000 \
    --name spectro-registry \
    -v $(pwd)/spectropaxconfig/:/etc/spectropaxconfig/ \
    gcr.io/spectro-images-public/release/spectro-registry:3.3.0  \
    serve /etc/spectropaxconfig/myconfig.yml
```

> Now you can access the pax registry at `https://yourhost.companydomain.com/v1/`.
> You will be prompted to give the user admin and the password of your choice.

Reviewer: ```suggestion
You can now access the pack registry at `https://yourhost.companydomain.com/v1/`.
You will be prompted to give the user admin and the password of your choice.
```

> # Deploying a Pack Registry Server with self signed certificates

Reviewer: ```suggestion
# Deploy a Pack Registry Server with Self-Signed Certificates
```
We avoid starting titles with verbs that end in -ing.

> The following steps need to be performed to deploy the pack registry server using self signed certificates:

Reviewer: ```suggestion
The following steps need to be performed to deploy the pack registry server using self-signed certificates:
```

7. Pull the latest Palette pack registry Docker image using the docker CLI.

    ```bash

>         docker pull gcr.io/spectro-images-public/release/spectro-registry:3.3.0

Reviewer: ```suggestion
 docker pull gcr.io/spectro-images-public/release/spectro-registry:3.3.0
```

7. Pull the latest Palette pack registry Docker image using the docker CLI.

    ```bash

>  docker pull gcr.io/spectro-images-public/release/spectro-registry:3.3.0

Author: nope. not here ...
